src/lib/y2partitioner/widgets/encrypt_password.rb

```diff
@@ -93,6 +73,94 @@ def pw1
       def pw2
         Yast::UI.QueryWidget(Id(:pw2), :Value)
       end
+
+      # @return [Boolean]
+      def random_password_allowed?
+        @random_password_allowed ||= @controller.filesystem_type.to_sym == :swap
+      end
+
+      # @return [Boolean]
+      def random_password?
+        return false unless random_password_allowed?
+
+        Yast::UI::QueryWidget(Id(:pwd_type), :Value) == :random_pwd
+      end
+
+      def form_content
+        if random_password_allowed?
+          extended_content
+        else
+          password_fields
+        end
+      end
+
+      def extended_content
+        VBox(
+          RadioButtonGroup(
+            Id(:pwd_type),
+            VBox(
+              Left(RadioButton(Id(:manual_pwd), _("Ask for password on boot to mount the swap"), true)),
+              password_fields,
+              Left(RadioButton(Id(:random_pwd), _("Auto-generate a new &random password on every boot"))),
```

Comment:
as I said above I expect this won't work well when going back as default is hardcoded here. Which I consider as future problem.